cirq-core/cirq/circuits/circuit.py

```diff
@@ -1712,12 +1712,26 @@ def __init__(
                 together. This option does not affect later insertions into the
                 circuit.
         """
-        self._moments: List['cirq.Moment'] = []
-        with _compat.block_overlapping_deprecation('.*'):
-            if strategy == InsertStrategy.EARLIEST:
-                self._load_contents_with_earliest_strategy(contents)
-            else:
-                self.append(contents, strategy=strategy)
+        self._moments: List[Moment] = []
+        if len(contents) == 1 and isinstance(contents[0], AbstractCircuit):
+            self._moments.extend(contents[0].moments)
+        elif all(isinstance(item, Moment) for item in contents):
+            self._moments.extend(cast(Iterable[Moment], contents))
+        else:
+            with _compat.block_overlapping_deprecation('.*'):
+                if strategy == InsertStrategy.EARLIEST:
+                    self._load_contents_with_earliest_strategy(contents)
+                else:
+                    self.append(contents, strategy=strategy)
+
+    @classmethod
+    def from_moments(cls, *moments: 'cirq.OP_TREE') -> 'Circuit':
+        """Create a circuit from moments.
+
+        Args:
+            *moments: Op tree for each moment.
+        """
+        return cls(Moment(moment) for moment in moments)
```

Comment:
Should it be `cls(*(Moment(moment) for moment in moments))`?
Right now it is taking the old branch on lines 1722-23.